python/Ganga/GPIDev/Lib/File/MassStorageFile.py

```diff
@@ -324,39 +353,37 @@ def put(self):
         if regex.search(fileName) is not None:
             for currentFile in glob.glob(os.path.join(sourceDir, fileName)):
                 finalFilename = filenameStructure.replace('{fname}', os.path.basename(currentFile))
-                (exitcode, mystdout, mystderr) = self.execSyscmdSubprocess('%s %s %s' % (cp_cmd, currentFile, os.path.join(massStoragePath, finalFilename)))
+                try:
+                    folder_ = os.path.dirname(os.path.join(massStoragePath, finalFilename))
+                    self._mkdir(folder_)
```

Comment:
This isn't needed as `finalFilename` is derived from an `os.basename` call so can't have a subdir prepended to it. Plus the `massStoragePath` has already been created further up.